### PR TITLE
Scroll chat to bottom on input focus. Closes #210

### LINKED ISSUE #210: Scroll chat to bottom after input focus

### PR DESCRIPTION
**Pull request by prastut**
Closes #210

### CODE REVIEW

**Comment by atd** [comment]
Thank you @prastut Just a couple of minor comments to keep the code clean (see the commit)

File: src/js/directives/chat.js [hunk 1]
```diff
@@ -53,6 +53,22 @@ angular.module('Teem')
             }
           };
 
+          //For scrolling to bottom on input focus
+          $scope.scrollToBottom = function(){
+            
+            var chatMessages = angular.element(document.getElementsByClassName('chat-message-text'));
+
+            if(chatMessages && chatMessages.length > 0){
```
Comment (atd) [inline src/js/directives/chat.js]: `chatMessages` will be defined even if it is empty. No need to check the first condition in the `if`

File: src/js/directives/chat.js [hunk 2]
```diff
@@ -53,6 +53,22 @@ angular.module('Teem')
...
+              
+              var lastMessage = chatMessages[chatMessages.length - 1];
+              var scrollableContentController = angular.element(lastMessage).controller('scrollableContent'); 
+
+              if(scrollableContentController){
```
Comment (atd) [inline src/js/directives/chat.js]: If you have reached this, `scrollableContentController` will be present. There shouldn't be a need for this `if`. Or I am missing something?

Comment (prastut) [inline src/js/directives/chat.js]: @atd Hmm true. I probably was thinking of an edge case while coding, now can't think of any. Nice observation. :+1:

File: src/js/directives/chat.js (hunk 1 above)
Comment (prastut) [inline src/js/directives/chat.js]: True @atd. Done.

**Comment by prastut** [comment]
@atd updated the code based on your comments. Will squash and rebase with master as soon as you approve them.

**Comment by atd** [comment]
@prastut awesome! Please go ahead :+1:

**Comment by prastut** [comment]
@atd Run into some trouble. I fetched upstream and merged it to my master. Then I checked out my branch for this issue and then rebased it and then pushed the code. 

What would be an appropriate way to resolve this issue? Want to know the right way. Otherwise I could just make a new branch, do the same changes again and push it which seems lame.

**Comment by atd** [comment]
@prastut what a mess :smiley: 

I would do the following:

`git checkout issue210` you will work on the issue branch
`git reset --hard b9f89a4aaccb` so you point the branch to your last commit
`git rebase p2pvalue/master` so you put your two commits on the top of our master
(this needs you to have a remote named "p2pvalue" pointing to p2pvalue/teem)
`git rebase -i HEAD~2` to squash the two commits into one https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/
`git push origin +issue210` to push the new commits to github

Hope it helps!

**Comment by prastut** [comment]
@atd earlier had done the same. I used to follow this methodology:

* `git checkout master`
* `git fetch upstream` where upstream points to p2pvalue/teem/master
* `git merge upstream\master`. Now my fork is levelled up with the remote.
* `git checkout issue210` followed by `git rebase master` 
* After that squash and push.

I used your strategy but wanted to know what was wrong with mine. Thanks for the help though. :+1:

**Comment by atd** [comment]
@prastut merges create one commit more if they cannot fast forward, so maybe you had some commits in master?

**Pull request merged**